tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -337,54 +298,102 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 				samples: [],
 				variance: Number.NaN,
 			};
+
 			do {
 				await testObject.beforeIteration?.();
-
 				global.gc();
-				benchmarkStats.samples.before.memoryUsage.push(process.memoryUsage());
-				benchmarkStats.samples.before.heap.push(v8.getHeapStatistics());
-				benchmarkStats.samples.before.heapSpace.push(v8.getHeapSpaceStatistics());
+				(
+					benchmarkStats.customData["Samples used"] as MemorySampleData
```

Comment:
Here we don't need all the data from the samples inside customData (we don't print it to console, and as we discussed we are ok losing it in the json file). So we can avoid all this casting (which is a red flag in the first place) and just use a local variable that is of the appropriate type. It's all "temporary working data" to calculate the totals that we do want to put in customData further down.

Reply:
Synced offline. We are not logging `samples`